app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/rejectedgoodsmultiple/EnterMovementReferenceNumberController.scala

```diff
@@ -0,0 +1,153 @@
+    extends RejectedGoodsMultipleJourneyBaseController
+    with SessionDataExtractor
+    with SessionUpdates
+    with Logging {
+
+  implicit val dataExtractor: DraftClaim => Option[MRN] = _.movementReferenceNumber
+
+  def show(): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    Future.successful {
+      val emptyForm = movementReferenceNumberForm
+      val form      = journey.getLeadMovementReferenceNumber.fold(emptyForm)(emptyForm.fill)
+      val router    = ReimbursementRoutes(JourneyBindable.Multiple)
+      Ok(
+        enterMovementReferenceNumberPage(
+          form,
+          router.subKey,
+          routes.EnterMovementReferenceNumberController.enterMrnSubmit()
+        )
+      )
+    }
+  }
+
+  def enterMrnSubmit(): Action[AnyContent] = actionReadJourney { implicit request => journey =>
+    movementReferenceNumberForm
+      .bindFromRequest()
+      .fold(
+        formWithErrors =>
+          BadRequest(
+            enterMovementReferenceNumberPage(
+              formWithErrors,
+              ReimbursementRoutes(JourneyBindable.Multiple).subKey,
+              routes.EnterMovementReferenceNumberController.enterMrnSubmit()
+            )
+          ),
+        mrnNumber => {
+          val isSameAsPrevious =
+            journey.answers.movementReferenceNumbers.map(mrnSeq => mrnSeq).exists(_.contains(mrnNumber))
+
+          if (isSameAsPrevious && journey.hasCompleteAnswers)
+            Future.successful(Redirect(routes.CheckYourAnswersController.show()))
+          else {
+            for {
+              maybeAcc14    <-
+                claimService.getDisplayDeclaration(mrnNumber).leftMap(_ => Error("Could not get declaration"))
+              updateJourney <- updateJourney(journey, mrnNumber, maybeAcc14)
+            } yield updateJourney
+          }.fold(
+            errors => {
+              logger.error(s"Unable to record $mrnNumber", errors.toException)
+              Redirect(baseRoutes.IneligibleController.ineligible())
+            },
+            updatedJourney => redirectLocation(updatedJourney)
+          )
+        }
+      )
+  }
+
+  private def updateJourney(
+    journey: RejectedGoodsMultipleJourney,
+    mrn: MRN,
+    maybeAcc14: Option[DisplayDeclaration]
+  ): EitherT[Future, Error, RejectedGoodsMultipleJourney] =
+    maybeAcc14 match {
+      case Some(acc14) =>
+        EitherT.fromEither[Future](
+          journey.submitMovementReferenceNumberAndDeclaration(mrn, acc14).left.map(Error.apply(_))
+        )
+      case _           =>
+        EitherT.leftT(Error("could not unbox display declaration"))
+    }
+
+  private def redirectLocation(journey: RejectedGoodsMultipleJourney): Result =
+    if (journey.needsDeclarantAndConsigneeEoriSubmission) {
+      Redirect(routes.EnterDeclarantEoriNumberController.show())
+    } else {
+      Redirect(
+        routes.WorkInProgressController.show() //TODO: "Update multiple/check-declaration-details route"))
```

Comment:
it already exists! you are the blocker

Reply:
@aneksamun Okay can you point me to the right routes entry? I didn't find one in `rejected_goods_multiple.routes` ...

Reply:
My bad, it is indeed in progress. Steve is working on it.